### PR TITLE
drakrun: Add support for custom apimon hooks

### PR DESCRIPTION
**Pull request by chivay**

### CODE REVIEW

**Comment by BonusPlay** [comment]
Imo if we're `overriding` default hooks, we should `override`, not `add more`.

**Comment by chivay** [comment]
Who said anything about overriding :thinking:

**Comment by BonusPlay** [comment]
Well, this "adds" more hooks, but it could very easily be turned into "override" hooks, which has ability to subtract them as well, thus being superior :)

**Comment by chivay** [comment]
:thinking: I guess

**Comment by icedevml** [comment]
I think that hook override (in a way that you reduce number of hooks) might be useful. For instance, if you have some very heavy stuff and you only want to monitor certain APIs to reduce overhead.

LGTM in general

**Comment by chivay** [comment]
Reworked to behave as an override mechanism instead of adding new entries

File: drakrun/drakrun/main.py [hunk 1]
```diff
@@ -326,8 +326,8 @@ def analysis_uid(self):
 
         return self.current_task.uid
 
-    def build_drakvuf_cmdline(self, timeout, cwd, full_cmd, dump_dir, ipt_dir):
-        hooks_list = os.path.join(ETC_DIR, "hooks.txt")
+    def build_drakvuf_cmdline(self, timeout, cwd, full_cmd, dump_dir, ipt_dir, workdir):
```
Comment (BonusPlay) [inline drakrun/drakrun/main.py:329]: If we add `workdir` param here, then surely we need to add an argument to where we call this function from.

Comment (chivay) [reply drakrun/drakrun/main.py:329]: we do, on line 471

Comment (BonusPlay) [reply drakrun/drakrun/main.py:329]: Ah, I'm blind.

**Review by BonusPlay — APPROVED** [review]
:wew:

**Pull request merged**